gen/bbmri_negotiator

```diff
@@ -0,0 +1,212 @@
+		$roleIdValue = $data->getGroupAttributeValue( group => $groupId, attrName => $A_ADM_COLLECTION_ID );
+		$groupRole = $GROUP_ROLE_ADM_COLLECTION;
+	} elsif ($data->getGroupAttributeValue( group => $groupId, attrName => $A_ADM_BIOBANK_ID )) {
+		$roleIdValue = $data->getGroupAttributeValue( group => $groupId, attrName => $A_ADM_BIOBANK_ID );
+		$groupRole = $GROUP_ROLE_ADM_BIOBANK;
+	} elsif ($data->getGroupAttributeValue( group => $groupId, attrName => $A_ADM_NATIONAL_NODE_ID )) {
+		$roleIdValue = $data->getGroupAttributeValue( group => $groupId, attrName => $A_ADM_NATIONAL_NODE_ID );
+		$groupRole = $GROUP_ROLE_ADM_NN;
+	} elsif ($data->getGroupAttributeValue( group => $groupId, attrName => $A_ADM_NETWORK_ID )) {
+		$roleIdValue = $data->getGroupAttributeValue( group => $groupId, attrName => $A_ADM_NETWORK_ID );
+		$groupRole = $GROUP_ROLE_ADM_NETWORK;
+	} else {
+		return;
+	}
+
+	for my $memberId ( $data->getMemberIdsForResourceAndGroup( resource => $resourceId, group => $groupId )) {
+		processUsers $groupId, $memberId, $roleIdValue, $groupRole;
+	}
+}
+
+sub initializeUser {
+	my ($uid, $memberId) = @_;
+	$userStruc->{$uid} = {};
+	my $email = $data->getUserAttributeValue( member => $memberId, attrName => $A_USER_EMAIL );
+	my $d_name = $data->getUserAttributeValue( member => $memberId, attrName => $A_USER_D_NAME );
+
+	$userStruc->{$uid}->{$u_email} = $email;
+	$userStruc->{$uid}->{$u_name} = $d_name;
+	$userStruc->{$uid}->{$u_collections} = [];
+	$userStruc->{$uid}->{$u_biobanks} = [];
+	$userStruc->{$uid}->{$u_national_nodes} = [];
+	$userStruc->{$uid}->{$u_networks} = [];
+	$userStruc->{$uid}->{$u_adm_collections} = [];
+	$userStruc->{$uid}->{$u_adm_biobanks} = [];
+	$userStruc->{$uid}->{$u_adm_national_nodes} = [];
+	$userStruc->{$uid}->{$u_adm_networks} = [];
+}
+
+sub processUsers {
+	my ($gid, $memberId, $roleIdValue, $groupRole) = @_;
+
+	my $uid = $data->getUserAttributeValue( member => $memberId, attrName => $A_LSRI_USER_ID );
+	unless(exists $userStruc->{$uid}) {
+		initializeUser $uid, $memberId; 
+	}
+
+	my $status = $data->getMemberAttributeValue( member => $memberId, attrName => $A_USER_STATUS );
+	my $isSuspended = $data->getMemberAttributeValue( member => $memberId, attrName => $A_MEMBER_IS_SUSPENDED );
+	if ($isSuspended) {
+		$status = $STATUS_SUSPENDED;
+	}
+	my $memberStatus = $userStruc->{$uid}->{$u_status} // $status;
+	if ($memberStatus eq $STATUS_EXPIRED && $status eq $STATUS_VALID){
+		# change from EXPIRED to VALID
+		$userStruc->{$uid}->{$u_status} = $status;
+	} elsif ($memberStatus eq $STATUS_SUSPENDED && $status eq $STATUS_VALID){
+		# change from SUSPENDED to VALID
+		$userStruc->{$uid}->{$u_status} = $status;
+	} elsif ($memberStatus eq $STATUS_SUSPENDED && $status eq $STATUS_EXPIRED){
+		# change from SUSPENDED to EXPIRED
+		$userStruc->{$uid}->{$u_status} = $status;
+	}
+	$userStruc->{$uid}->{$u_status} = $status;
```

Comment:
```suggestion
$userStruc->{$uid}->{$u_status} = $status;
```
Am I missing something? Maybe intended to be extended in the future?

Reply:
Copy pasta from the bbmri_collections and bbmri_networks scripts. Not sure why this was implemented this way. But actually, we are only interested in the membership status in the BBMRI VO. It is also assumed that resources will be only for the BBMRI VO. So I think the whole mechanism can be removed (I am not even sure if the end service will care about the status).

Reply:
alright, we can then filter the non VALID members and not include the status in the result.